crates/sui-core/src/authority_aggregator.rs

```diff
@@ -198,16 +251,47 @@ struct ProcessTransactionState {
     effects_map: MultiStakeAggregator<TransactionEffectsDigest, TransactionEffects, true>,
     // The list of errors gathered at any point
     errors: Vec<(SuiError, Vec<AuthorityName>, StakeUnit)>,
-    bad_stake: StakeUnit,
+    // This is exclusively non-retryable stake.
+    non_retryable_stake: StakeUnit,
+    // This includes both object and package not found sui errors.
+    object_or_package_not_found_stake: StakeUnit,
     // If there are conflicting transactions, we note them down and may attempt to retry
     conflicting_tx_digests:
         BTreeMap<TransactionDigest, (Vec<(AuthorityName, ObjectRef)>, StakeUnit)>,
+    // As long as none of the exit criteria are met we consider the state retryable
+    // 1) >= 2f+1 signatures
+    // 2) >= f+1 non-retryable errors
+    // 3) >= 2f+1 object not found errors
```

Comment:
why do we differentiate the quorum needed between case 2 and case 3? malicious validators can send case 2 or case 3 messages so why the differentiation of how many honest replies we want? 

I am mostly not sure why 2f+1 in case 3)

Reply:
We special cased object/package not found cases to not immediately consider them non-retryable. As long as 2f+1 validators have not said they don't have the objects it is possible that eventually those objects become available to the validators on a retry.